projects/go-lib/src/lib/components/go-file-upload/go-file-upload.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+import { AbstractControl, FormArray, FormBuilder, FormControl, FormGroup } from '@angular/forms';
+
+@Component({
+  selector: 'go-file-upload',
+  templateUrl: './go-file-upload.component.html',
+  styleUrls: ['./go-file-upload.component.scss']
+})
+export class GoFileUploadComponent implements OnInit {
+  form: FormGroup;
+  files: FormArray;
+  fb: FormBuilder;
+  filePreview: Array<string> = [];
+  id: string;
+
+  @Input() control: FormControl;
+  @Input() hints: Array<string> = [];
+  @Input() isLoading: boolean = false;
+  @Input() key: string;
+  @Input() label: string;
+  @Input() multiple: boolean = false;
+  @Input() state: 'selecting' | 'selected' = 'selecting';
+  @Input() theme: 'light' | 'dark' = 'light';
```

Comment:
This binding exists, but it doesn't seem this actually works as intended. Can you take a look at the styling for a `dark` theme and update them to be correct please?

Reply:
We'll also want documentation for this.

Reply:
I fixed the dark theming here and added documentation for it.